lib/Remotery.c

```diff
@@ -4266,7 +4302,7 @@ Message* rmtMessageQueue_PeekNextMessage(rmtMessageQueue* queue)
     // the next one in the queue is ready.
     r = r & (queue->size - 1);
     ptr = (Message*)(queue->data->ptr + r);
-    id = (MessageID)LoadAcquire((rmtU32*)&ptr->id);
+    id = (MessageID)LoadAcquire((rmtAtomicU32*)&ptr->id);
```

Comment:
Does this compile on C++? I can't see a `LoadAcquire` that matches the pointer type.

Reply:
Well, it compiled as before but ran with asan warnings.
I now also added the stdatomic.h for c++11.

Reply:
So now, with
```
clang -std=c11
clang++ -std=c++11
```
, there aren't any TSAN warnings.

With `clang -std=c99`, there are.
For C++ I don't get any warnings with `clang++ -std=c++98` either.

Reply:
Oh I see, it's because there are no atomics on the C++ version.